### PR TITLE
updated dof validation to use multiple cameras, added support for dof…

### PR DESCRIPTION
**Pull request by TrevorDev**
…/circle of confusion to specify a camera to be used

https://github.com/BabylonJS/Babylon.js/issues/3764

### CODE REVIEW

File: src/PostProcess/babylon.circleOfConfusionPostProcess.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ module BABYLON {
         /**
          * Creates a new instance of @see CircleOfConfusionPostProcess
          * @param name The name of the effect.
-         * @param scene The scene the effect belongs to.
+         * @param minMaxZCamera The camera to be used to calculate the min/max z for the depth map
```
Comment (TrevorDev) [inline src/PostProcess/babylon.circleOfConfusionPostProcess.ts]: Now this constructor takes 2 camera parameters which is kind of weird, but the second one must be null when used in the pipeline so I think this needs to be done.

File: src/PostProcess/RenderPipeline/Pipelines/babylon.defaultRenderingPipeline.ts [hunk 2]
```diff
@@ -286,7 +286,7 @@
             this._reset();
 
             if(this.depthOfFieldEnabled){
-                this.depthOfField = new DepthOfFieldEffect(this._scene, this._depthOfFieldBlurLevel, this._defaultPipelineTextureType);
+                this.depthOfField = new DepthOfFieldEffect(this._scene, this._cameras[0], this._depthOfFieldBlurLevel, this._defaultPipelineTextureType);
```
Comment (deltakosh) [inline src/PostProcess/RenderPipeline/Pipelines/babylon.defaultRenderingPipeline.ts]: I think we should  not force cameras[0] but use the pipeline mechanism about active cameras

Comment (TrevorDev) [reply src/PostProcess/RenderPipeline/Pipelines/babylon.defaultRenderingPipeline.ts]: If the scene has multiple cameras rendering at the same time, I dont want the second camera to get the firsts depth map.

File: src/PostProcess/babylon.circleOfConfusionPostProcess.ts (hunk 1 above)
Comment (deltakosh) [reply src/PostProcess/babylon.circleOfConfusionPostProcess.ts]: I would prefer saving min and max as property of the depth texture as we know them when we create the depth texture

Comment (TrevorDev) [reply src/PostProcess/babylon.circleOfConfusionPostProcess.ts]: Sounds good, changed circle of confusion and depth of field to take cameraMinZ:number, cameraMaxZ:number, depthTexture: RenderTargetTexture as parameters instead

File: src/PostProcess/babylon.circleOfConfusionPostProcess.ts [hunk 3]
```diff
@@ -32,7 +33,7 @@ module BABYLON {
          * @param reusable If the post process can be reused on the same frame. (default: false)
          * @param textureType Type of textures used when performing the post process. (default: 0)
          */
-        constructor(name: string, scene: Scene, depthTexture: RenderTargetTexture, options: number | PostProcessOptions, camera: Nullable<Camera>, samplingMode?: number, engine?: Engine, reusable?: boolean, textureType: number = Engine.TEXTURETYPE_UNSIGNED_INT) {
+        constructor(name: string, cameraMinZ:number, cameraMaxZ:number, depthTexture: RenderTargetTexture, options: number | PostProcessOptions, camera: Nullable<Camera>, samplingMode?: number, engine?: Engine, reusable?: boolean, textureType: number = Engine.TEXTURETYPE_UNSIGNED_INT) {
```
Comment (deltakosh) [inline src/PostProcess/babylon.circleOfConfusionPostProcess.ts]: Actually I meant to literally read these values on the depthTexture with the depthTexture.camera. So not even need to give these values to the constructor

**Pull request merged**